homeassistant/components/dovado/sensor.py

```diff
@@ -0,0 +1,118 @@
+"""
+Support for Dovado router.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.dovado/
+"""
+import logging
+import re
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.dovado import DOMAIN as DOVADO_DOMAIN
+from homeassistant.components.sensor import (PLATFORM_SCHEMA)
+from homeassistant.const import CONF_SENSORS
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['dovado']
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+SENSOR_UPLOAD = 'upload'
+SENSOR_DOWNLOAD = 'download'
+SENSOR_SIGNAL = 'signal'
+SENSOR_NETWORK = 'network'
+SENSOR_SMS_UNREAD = 'sms'
+
+SENSORS = {
+    SENSOR_NETWORK: ('signal strength', 'Network', None,
+                     'mdi:access-point-network'),
+    SENSOR_SIGNAL: ('signal strength', 'Signal Strength', '%',
+                    'mdi:signal'),
+    SENSOR_SMS_UNREAD: ('sms unread', 'SMS unread', '',
+                        'mdi:message-text-outline'),
+    SENSOR_UPLOAD: ('traffic modem tx', 'Sent', 'GB',
+                    'mdi:cloud-upload'),
+    SENSOR_DOWNLOAD: ('traffic modem rx', 'Received', 'GB',
+                      'mdi:cloud-download'),
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SENSORS): vol.All(
+        cv.ensure_list, [vol.In(SENSORS)]
+    ),
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Dovado sensor platform."""
+    dovado = hass.data[DOVADO_DOMAIN]
+
+    entities = []
+    for sensor in SENSORS:
+        if sensor in config.get[CONF_SENSORS]:
+            entities.append(DovadoSensor(dovado, sensor))
+
+    add_entities(entities)
+    return True
```

Comment:
Nothing is checking this return value.

Reply:
I thought setup_platform and async_setup_platform always had to return a true/false to indicate a correct/incorrect setup. Is that understanding incorrect?